### PR TITLE
Don't throw when grabbing injected provider

### PR DESCRIPTION
**Pull request by fan-zhang-sv**
### _Summary_

window.top may throw DOMException, we should not let this error blocking us from continuing

### CODE REVIEW

**Review by nateReiners — APPROVED** [review]
lgtm

**Comment by cb-heimdall** [comment]
Review Error for nateReiners @ 2024-06-21 20:31:14 UTC
User failed mfa authentication, public email is not set on your github profile. see [go/mfa-help](http://go/mfa-help)

File: packages/wallet-sdk/src/util/provider.ts [hunk 1]
```diff
@@ -61,10 +61,13 @@ export function getCoinbaseInjectedProvider({
     }
   }
 
-  const ethereum = window.ethereum ?? window.top?.ethereum;
```
Comment (bangtoven) [inline packages/wallet-sdk/src/util/provider.ts:64]: how about creating a dedicated function or something to do just this and catch error?
the if statement below doesn’t throw error anyway.

**Review by nateReiners — APPROVED** [review]

**Review by bangtoven — COMMENTED** [review]
@fan-zhang-sv

**Pull request merged**